api/v1/platformnetwork_webhook.go

```diff
@@ -42,56 +33,6 @@ func (r *PlatformNetwork) Default() {
 	// TODO(user): fill in your defaulting logic.
 }
 
-// Determines if a string is a valid IP address
-func IsIPAddress(value string) bool {
-	return common.IsIPv4(value) || common.IsIPv6(value)
-}
-
-// Determines if the a prefix length agrees with the address family of the specified address
-func IsValidPrefix(address string, prefix int) bool {
-	if common.IsIPv4(address) {
-		if prefix <= MaxIPv4PrefixLength {
-			return true
-		}
-	} else if common.IsIPv6(address) {
-		if prefix <= MaxIPv6PrefixLength {
-			return true
-		}
-	}
-	return false
-}
-
-// Validates an incoming resource update/create request.  The intent of this validation is to perform only the
-// minimum amount of validation which should normally be done by the CRD validation schema, but until kubebuilder
-// supports the necessary validation annotations we need to do this in a webhook.  All other validation is left
-// to the system API and any errors generated by that API will be reported in the resource status and events.
-// Validates that all address specifications within the network are of the same address family.
-func (r *PlatformNetwork) validatePlatformNetwork() error {
```

Comment:
The removal of validatePlatformNetwork is unpleasant. However, the original validation only validates the address pool values. I'm fine with it. May need to consider if other validations, can you add (TODO) in the following ValidateCreate ValidateUpdate methods

Reply:
It has been added [here](https://github.com/Wind-River/cloud-platform-deployment-manager/pull/350/files#diff-5ccf80dc481f9473df3b4d9364346c7b872893b7002e4c10ff0bc4dc7e508eaeR41
) and [here](https://github.com/Wind-River/cloud-platform-deployment-manager/pull/350/files#diff-5ccf80dc481f9473df3b4d9364346c7b872893b7002e4c10ff0bc4dc7e508eaeR50).

Reply:
Why not validate the associated networks?

Reply:
@hugonicodemos currently we don't have any special validations in mind for platformnetworks other than kubebuilder validations that are in place.
As @yjian118 has pointed out, I've added a TODO on myself to identify such validations to be done if at all any.